shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorker.java

```diff
@@ -295,6 +299,11 @@ public static TaskConfiguration buildTaskConfig(final PipelineConfiguration pipe
     private boolean isUncompletedJobOfSameSchemaInJobList(final String schema) {
         boolean isUncompletedJobOfSameSchema = false;
         for (JobInfo each : PipelineJobAPIFactory.getRuleAlteredJobAPI().list()) {
+            List<JobProgress> unFinishedJob = PipelineJobAPIFactory.getRuleAlteredJobAPI().getProgress(each.getJobId()).values().stream()
+                    .filter(progress -> !JobStatus.FINISHED.equals(progress.getStatus())).collect(Collectors.toList());
```

Comment:
Use `stream().allMatch` could be better, `List<JobProgress> unFinishedJob` is not necessary.